### PR TITLE
Doc 1778

### PR DESCRIPTION
**Pull request by regit**
Add documentation about IPS running modes.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/1778

Describe changes:
- AF_PACKET IPS mode
- Nftables rules examples
- Advanced option of nfqueue

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/499
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/275

### CODE REVIEW

**Comment by victorjulien** [comment]
Looks like you addressed https://redmine.openinfosecfoundation.org/issues/2876 as well, thanks!

File: doc/userguide/setting-up-ipsinline-for-linux.rst [hunk 1]
```diff
@@ -115,12 +118,134 @@ There is also a way to use iptables with multiple networks (and interface cards)
...
+    copy-mode: ips
+    copy-iface: eth0
+
+There is some important points to consider when setting up this mode:
+
+- The implementation of this mode is dependent of the zero copy mode of AF_PACKET. Thus you need to set `use-mmap` to yes on both interface.
+- MTU on both interfaces have to be equal: the copy from one interface to the other is direct and packet bigger then the MTU will be dropped by kernel.
+- Set different values of cluster-id on both interfaces to avoid conflict.
+- Any network card offloading creating bigger then physical layer datagram will result in packets drop as transmit path can not handle them.
+- Set `stream.inline` to `yes` so Suricata really switches to blocking mode
```
Comment (victorjulien) [inline doc/userguide/setting-up-ipsinline-for-linux.rst]: Ppl can leave it at 'auto', which is the default.

File: doc/userguide/setting-up-ipsinline-for-linux.rst [hunk 2]
```diff
@@ -115,12 +118,134 @@ There is also a way to use iptables with multiple networks (and interface cards)
...
+
+    copy-mode: ips
+    copy-iface: eth0
+
+There is some important points to consider when setting up this mode:
+
+- The implementation of this mode is dependent of the zero copy mode of AF_PACKET. Thus you need to set `use-mmap` to yes on both interface.
+- MTU on both interfaces have to be equal: the copy from one interface to the other is direct and packet bigger then the MTU will be dropped by kernel.
+- Set different values of cluster-id on both interfaces to avoid conflict.
+- Any network card offloading creating bigger then physical layer datagram will result in packets drop as transmit path can not handle them.
```
Comment (victorjulien) [inline doc/userguide/setting-up-ipsinline-for-linux.rst]: Maybe add: the common offloads that cause this are disabled by default.

File: doc/userguide/setting-up-ipsinline-for-linux.rst (hunk 1 above)
Comment (regit) [reply doc/userguide/setting-up-ipsinline-for-linux.rst]: outch, forgot about this code in AFP...

**Pull request closed without merging**